File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/singlepodcast/model/SinglePodcastModel.java
package com.pietrantuono.podcasts.addpodcast.singlepodcast.model;


import com.pietrantuono.podcasts.addpodcast.model.pojos.Podcast;
import com.pietrantuono.podcasts.apis.PodcastFeed;

import rx.Observer;

public interface SinglePodcastModel {

    void subscribeToFeed(Observer<PodcastFeed> observer);

    void unsubscribe();

    void subscribeToIsSubscribedToPodcast(Observer<Boolean> observer);

    void startModel(Podcast podcast);

    void onSubscribeUnsubscribeToPodcastClicked();
}

File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/singlepodcast/model/SinglePodcastModel.kt
package com.pietrantuono.podcasts.addpodcast.singlepodcast.model



import com.pietrantuono.podcasts.apis.PodcastFeed
import pojos.Podcast

import rx.Observer

interface SinglePodcastModel {

    fun subscribeToFeed(observer: Observer<PodcastFeed>)

    fun unsubscribe()

    fun subscribeToIsSubscribedToPodcast(observer: Observer<Boolean>)

    fun startModel(podcast: Podcast?)

    fun onSubscribeUnsubscribeToPodcastClicked()
}
